src/main/java/ca/uhn/fhir/jpa/starter/BackendAuthorizationInterceptor.java

```diff
@@ -104,4 +109,55 @@ private RSAPublicKey getKey() throws NoSuchAlgorithmException, InvalidKeySpecExc
         RSAPublicKeySpec publicKeySpec = new RSAPublicKeySpec(n, e);
         return (RSAPublicKey) kf.generatePublic(publicKeySpec);
     }
-}
\ No newline at end of file
+
+    /**
+     * Verify the access token signature and expiration
+     * @param token - the access token
+     * @return true if signature is valid and token has not expired, otherwise throws an exception
+     * @throws ParseException
+     */
+    private boolean verify(String token) throws IllegalArgumentException, NoSuchAlgorithmException, InvalidKeySpecException, TokenExpiredException, JWTVerificationException {
+        Algorithm algorithm = getAlgorithm(token, getKey());
+        JWTVerifier verifier = JWT.require(algorithm).build();
+        verifier.verify(token);
+
+        return true;
+    }
+
+    private Algorithm getAlgorithm(String token, Object publicKey) {
+        // Decode the header of the token
+        String header = token.split("\\.")[0];
+        byte[] decodedBytes = Base64.getDecoder().decode(header);
+        String decodedHeader = new String(decodedBytes);
+
+        // Get the alg
+        JSONObject headerJSON = new JSONObject(decodedHeader);
+        String alg = headerJSON.getString("alg");
+
+        switch(alg) {
+            case "HS256":
+                return Algorithm.HMAC256((String) publicKey);
+            case "HS384":
+                return Algorithm.HMAC384((String) publicKey);
+            case "HS512":
+                return Algorithm.HMAC512((String) publicKey);
+            case "RS256":
+                return Algorithm.RSA256((RSAPublicKey) publicKey, null);
+            case "RS384":
+                return Algorithm.RSA384((RSAPublicKey) publicKey, null);
+            case "RS512":
+                return Algorithm.RSA512((RSAPublicKey) publicKey, null);
+            case "ES256":
+                return Algorithm.ECDSA256((ECPublicKey) publicKey, null);
+            case "ES384":
+                return Algorithm.ECDSA384((ECPublicKey) publicKey, null);
+            case "ES512":
+                return Algorithm.ECDSA512((ECPublicKey) publicKey, null);
+            case "PS256":
+            case "PS384":
+            default:
+                return null;
```

Comment:
Rather than just return null for these this seems to be an appropriate usage of `NoSuchAlgorithmException`: "This exception is thrown when a particular cryptographic algorithm is requested but is not available in the environment."  https://docs.oracle.com/javase/8/docs/api/java/security/NoSuchAlgorithmException.html

Otherwise if we return null here, the following JWT.verify call will eventually `throw new IllegalArgumentException("The Algorithm cannot be null.");` which doesn't make much sense